reference/activity_manager/project_completeness.rst

```diff
@@ -0,0 +1,111 @@
+=================================
+
+Overview
+________
+
+In this reference guide you will learn what the project completeness refers to and how we manage it in Akeneo PIM. It
+will explain our technical and structure choices.
+
+Goal
+____
+
+The Activity Manager feature is about creating project from a products selection that give a vision on the enrichment
+progress to contributors (users in charge of the enrichment). Each project gives to the contributor its own completeness
+on the products selection, taking into account its permissions to edit or not products, attribute groups, locales,
+scopes.
+
+For example, if Bob can only edit the "Technical" attribute group of the "High-tech" category, he will see the project
+100% complete if all attributes in the attribute group "Technical" of "High-tech" products in the project are filled,
+even if other attribute groups are empty and required by the family. However, Julia, the owner of the project, will see
+that it is not complete if some attributes are not filled in others attribute groups. The contributor see the
+completeness of the project in terms of his own permissions, whereas the owner see the completeness of all the project.
+
+As every user will have his very own completeness depending on his permissions, we need to walk through every products
+to compute completeness for every user. As you can guess, it's a really huge task that cannot be done synchronously.
+
+We need to go deeper
+____________________
+
+We found two ways to deal with this problem. The first one was to calculate this detailed completeness for each user and
+cache it in a table. The benefit of this method is clear, each time we need those numbers we simply query the database:
+"Hey you, I just met Mary, what is her completeness? TODO: 42, IN_PROGRESS: 1251, DONE: 3254.". And this is crazy,
+result is instant, but the main drawback, is that the data is not up to date as this table is a cache. It needs to be
+calculated as often as possible but the task is so huge that with many different projects and users a night could not
+be sufficient.
+
+One table to rule them all
+__________________________
+
+This first method seems not to be the chosen one. Let's recap which data we need:
+ - For each locale,
+ - For each channel,
+ - For each user,
+ - Does the product
+ - have at least an attribute filled in?
+ - or is it complete?
+
+Oh. Can you see this?
+
+Doing a table to gather all data we need to compute the completeness seems to be a good idea but we need some
+adjustments. It can't be scalable if we do this for each user.
+
+.. _format: format.html
+
+So let's remove users from this table. And now, how to retrieve the completeness from a user? In the project format_,
+we have product filters. From those filters it's easy to know which user is impacted by the products selection. To win
+some time, we pre-calculate this data linking users impacted to project at its creation. But it's not scalable either.
```

Comment:
I don't understand
"pre-calculate this data linking users impacted to project at its creation"